### PR TITLE
Add game rule to allow repeated species

### PR DESCRIPTION
**Pull request by Oberlus**
I considered making it an integer (1 for no repetitions, 2+ for repetitions) but I guess this is enough and nicer in UI.

### CODE REVIEW

File: default/python/universe_generation/empires.py [hunk 1]
```diff
@@ -46,9 +46,11 @@ def get_starting_species_pool():
...
     Empire species pool generator, return random empire species and ensure somewhat even distribution
     """
-    # fill the initial pool with one set of all playable species
-    # species won't repeat unless there are more players than playable species
-    pool = fo.get_playable_species()
+    # fill the initial pool of playable species, without repetitions unless RULE_ALLOW_REPEATED_SPECIES is true
+    if not fo.getGameRules().getBool("RULE_ALLOW_REPEATED_SPECIES"):
+        pool = fo.get_playable_species()
+    else:
+        pool = fo.get_playable_species() * 2
```
Comment (Oberlus) [inline default/python/universe_generation/empires.py:53]: I wasn't sure of this. Should it be like this?:
```
pool = fo.get_playable_species() * (1 + fo.getGameRules().getBool("RULE_ALLOW_REPEATED_SPECIES"))
```

Comment (geoffthemedio) [reply default/python/universe_generation/empires.py:53]: I think the if-else variant is clearer

**Comment by Oberlus** [comment]
This is ready for merge. The unsuccessful checks seem unrelated. Playtested in Ubuntu 20.04.

**Comment by Oberlus** [comment]
I don't understand the tests, but I found something weird:
https://github.com/freeorion/freeorion/runs/5599960967?check_suite_focus=true#step:13:60
In all those Ubuntu and Windows tests, at the beginning there is the processing of the game rules, and these do not include the one added in this PR, RULE_ALLOW_REPEATED_SPECIES.

**Comment by o01eg** [comment]
It doesn't have `getBool` method:

```
2022-03-18T12:00:14.5811459Z 3: 12:00:14.581043 {0x00007f680a5a4740} [error] server : PythonCommon.cpp:101 : Traceback (most recent call last):
2022-03-18T12:00:14.5812256Z 3: 12:00:14.581080 {0x00007f680a5a4740} [error] server : PythonCommon.cpp:101 :   File "/home/runner/work/freeorion/freeorion/build/default/python/common/listeners.py", line 40, in wrapper
2022-03-18T12:00:14.5812961Z 3:     res = funct(*args)
2022-03-18T12:00:14.5813677Z 3: 12:00:14.581087 {0x00007f680a5a4740} [error] server : PythonCommon.cpp:101 :   File "/home/runner/work/freeorion/freeorion/build/default/python/universe_generation/universe_generator.py", line 156, in create_universe
2022-03-18T12:00:14.5814242Z 3:     if not setup_empire(empire, psd.empire_name, home_system, psd.starting_species, psd.player_name):
2022-03-18T12:00:14.5814782Z 3: 12:00:14.581092 {0x00007f680a5a4740} [error] server : PythonCommon.cpp:101 :   File "/home/runner/work/freeorion/freeorion/build/default/python/universe_generation/empires.py", line 503, in setup_empire
2022-03-18T12:00:14.5815244Z 3:     starting_species = next(starting_species_pool)
2022-03-18T12:00:14.5815808Z 3: 12:00:14.581097 {0x00007f680a5a4740} [error] server : PythonCommon.cpp:101 :   File "/home/runner/work/freeorion/freeorion/build/default/python/universe_generation/empires.py", line 50, in get_starting_species_pool
2022-03-18T12:00:14.5816311Z 3:     if not fo.getGameRules().getBool("RULE_ALLOW_REPEATED_SPECIES"):
2022-03-18T12:00:14.5817086Z 3: 12:00:14.581102 {0x00007f680a5a4740} [error] server : PythonCommon.cpp:101 : AttributeError: 'GameRules' object has no attribute 'getBool'
```

**Comment by o01eg** [comment]
It called `getToggle`

https://github.com/freeorion/freeorion/blob/0590e073703eae52aaf6e8de2f33ebdb84cb0bf4/python/UniverseWrapper.cpp#L755

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -1210,6 +1210,12 @@ Enable Super-Tester Takeover
 RULE_ENABLE_SUPER_TESTER_DESC
 Enables [[BLD_SUPER_TEST]] building
 
+RULE_ALLOW_REPEATED_SPECIES
+Allows repeated species
+
+RULE_ALLOW_REPEATED_SPECIES_DESC
+Allows repeated species when there are less players than playable species
```
Comment (Cjkjvfnby) [inline default/stringtables/en.txt]: Multiplying the number of species on 2 still may not fix this. 

Maybe we should change the code in a way that this expression will be true `len(specias_to_shoose) * n > len(players)`. Where n is multiplier from 1 to inf

Comment (o01eg) [reply default/stringtables/en.txt]: Species pool automatically populates while assign to empire:

https://github.com/freeorion/freeorion/blob/0590e073703eae52aaf6e8de2f33ebdb84cb0bf4/default/python/universe_generation/empires.py#L57-L62

Comment (Cjkjvfnby) [reply default/stringtables/en.txt]: Got it.  The problem I described has already been solved. 

The description is still a bit confusing.

Maybe change it to
```
Allows repeated species, if the number of players is more than the number of species this option is turned on by default.
```

Comment (Oberlus) [reply default/stringtables/en.txt]: I think current comments are pretty specific and unambiguous:

> fill the initial pool of playable species, without repetitions unless RULE_ALLOW_REPEATED_SPECIES is true

The initial pool of playable species won't have repeated species unless the game rule is 1.

**Comment by Oberlus** [comment]
> It called `getToggle`

Thank you. I tried it and the rule is now parsed. Waiting for the tests to finish.

File: default/stringtables/en.txt (hunk 2 above)
Comment (Cjkjvfnby) [reply default/stringtables/en.txt]: This is not about the comment in code, this is about string tables entry.

Comment (Oberlus) [reply default/stringtables/en.txt]: Gah!

OK. Your suggestion seems also confusing to me. The option is not enabled when there are more players than species, it just refills itself when needed regardless of the option. So I tried a similar wording:

`Allows repeated species when there are less players than playable species. When there are more players than playable species there are always repeated species.`

The second sentence is redundant in my opinion, but I have no trouble leaving it there (it's only for the tooltip).

Comment (Cjkjvfnby) [reply default/stringtables/en.txt]: > The second sentence is redundant in my opinion, but I have no trouble leaving it there (it's only for the tooltip).

The refill feature is documented in the code. So regular players cannot read it.

Comment (Oberlus) [reply default/stringtables/en.txt]: So OK now?

**Comment by Oberlus** [comment]
Unless @Cjkjvfnby has anything pending of review, this is good to merge.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**